### PR TITLE
Trivial: Fix warning introduced by #7053 by casting to uint64_t

### PR DESCRIPTION
**Pull request by jtimon**

### CODE REVIEW

**Comment by laanwj** [comment]
This is inconsistent too:
```
src/chainparams.h:    int64_t PruneAfterHeight() const { return nPruneAfterHeight; }
src/chainparams.h:    uint64_t nPruneAfterHeight;
```

**Comment by jtimon** [comment]
I don't even really know why this parameter has to be in chainparams...

**Comment by jtimon** [comment]
Solved nit.

**Comment by jtimon** [comment]
What about this last commit? Maybe making it a configurable option or something?

**Comment by paveljanik** [comment]
Your last commit makes it 100000 in all networks. Do we want it?

**Comment by jtimon** [comment]
Do we don't? As said another option is making it configurable, but it really doesn't make sense to me that this is a per-chain thing.

**Comment by paveljanik** [comment]
I think making it 100000 for all networks is OK.

ACK (warning is gone, nGlobalPruneAfterHeight can be made configurable in the future if needed).

File: src/main.cpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ using namespace std;
 
 CCriticalSection cs_main;
 
+uint64_t nGlobalPruneAfterHeight = 100000;
```
Comment (MarcoFalke) [inline src/main.cpp]: Please fix doc of https://github.com/jtimon/bitcoin/blob/7992d97fa4e92351f9f4fa155384ea2181b28f56/src/main.h#L218

**Comment by paveljanik** [comment]
I'd remove the removal now to get this integrated...

**Comment by jtimon** [comment]
Yes, I can leave it at the first commit for now and leave the second for another PR, but I was curious to see how much code it would be and what people would think about it.

**Comment by paveljanik** [comment]
I think others did not read it at all now (because of Trivial etc.).

**Comment by sdaftuar** [comment]
The `nPruneAfterHeight` is different for each chain so that the pruning code can be tested.  In particular, changing the regtest value to be 100000 will break the `pruning.py` rpc test.

**Comment by MarcoFalke** [comment]
I am ok with restoring the initial commit, but please still fix Please fix doc of https://github.com/jtimon/bitcoin/blob/7992d97fa4e92351f9f4fa155384ea2181b28f56/src/main.h#L218 which says "10" for regtest.

**Comment by jtimon** [comment]
@MarcoFalke I will correct the doc, if you have a complete text replacement I can use that.

@sdaftuar Thanks for the clarification but it doesn't seem like the only possible way to test something (ie there's many other things that are testable and configurable without being in CChainParams.
I mean, if the RPC need a lower value I guess it's simpler to make it directly configurable than to adapt the tests.

**Comment by MarcoFalke** [comment]
```diff
<  * Pruning cannot take place until the longest chain is at least a certain length (100000 on mainnet, 1000 on testnet, 10 on regtest).
---
>  * Pruning cannot take place until the longest chain is at least a certain length (100000 on mainnet, 1000 on testnet, 1000 on regtest).

**Comment by jtimon** [comment]
Left only the most trivial things (including @MarcoFalke 's nit) and left the rest in https://github.com/bitcoin/bitcoin/compare/master...jtimon:chainparams-prune-out for now.

**Comment by paveljanik** [comment]
ACK

**Comment by MarcoFalke** [comment]
utACK  cb491e7

**Comment by laanwj** [comment]
> I mean, if the RPC need a lower value I guess it's simpler to make it directly configurable than to adapt the tests.

I agree. If that is the rationale, the RPC test could just pass in a hidden `-pruneafterheight` option, that seems to be better than making it dependent on the chain.

But yes let's not do it in this pull which proclaims to be a 'trivial' integer signedness fix.

**Pull request merged**